src/mobile-ui-react/ModalEntryFormDialog.tsx

```diff
@@ -169,19 +170,27 @@ export function ModalDialog(props: ModalDialogProps) {
     }, 0);
   }, []);
 
-  const handleOK = async () => {
+  const handleOK = React.useCallback(async () => {
+    if (waitingForOK)
+      return false;
+    setWaitingForOK(true);
     if (await onOK()) {
       setFadedOut(true);
       return true;
     }
+    setWaitingForOK(false);
```

Comment:
Maybe this should be put in a try/finally block so setWaitingForOK(false) is called when the user presses OK too? (Just so it's in the correct state)

Reply:
The current setup is intentional. The OK button immediately disables when the user hits it. By having this here, it stays disabled until the dialog is finished fading out.

Reply:
Sorry, I meant this comment to refer to the `if` block.

Something like this:
```ts
    try {
      if (await onOK()) {
        setFadedOut(true);
        return true;
      }
    } finally {
      setWaitingForOK(false);
    }
```

This way `setWaitingForOK(false)` is always called. I guess it doesn't really need to be called after the `onOK` handler has completed, but thought it might make sense for completeness.

Reply:
I didn't really intend to allow exceptions in `onOK`, but dealing with it like you suggest doesn't hurt anything, so I made the change.